### PR TITLE
fix: disable clippy for tests

### PR DESCRIPTION
**Pull request by niklasad1**
Because of https://github.com/rust-lang/rust-clippy/issues/9922 let's disable clippy for tests until that is fixed.

### CODE REVIEW

**Review by alvicsam — APPROVED** [review]

**Review by lexnv — APPROVED** [review]
LGTM! I'm wondering if we were to apply `clippy --fix` is it would still crash the clippy? 🤔

**Review by jsdw — APPROVED** [review]

File: proc-macros/Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ proc-macro = true
 [dependencies]
 proc-macro2 = "1.0"
 quote = "1.0"
-syn = { version = "1.0", default-features = false, features = ["extra-traits", "full", "visit", "parsing"] }
+syn = { version = "1.0", default-features = false, features = ["extra-traits", "full", "visit", "parsing", "printing", "clone-impls", "proc-macro"] }
```
Comment (jsdw) [inline proc-macros/Cargo.toml:18]: Interesting!

**Comment by niklasad1** [comment]
> LGTM! I'm wondering if we were to apply clippy --fix is it would still crash the clippy? thinking

Yes, I tried earlier :(

File: proc-macros/Cargo.toml (hunk 1 above)
Comment (niklasad1) [reply proc-macros/Cargo.toml:18]: These features leaked from the workspace but not sure why it's was catched now

**Pull request merged**